src/theory/strings/theory_strings_rewriter.cpp

```diff
@@ -984,6 +1073,40 @@ RewriteResponse TheoryStringsRewriter::postRewrite(TNode node) {
   {
     retNode = rewriteStringCode(node);
   }
+  else if (node.getKind() == REGEXP_CONCAT)
```

Comment:
Can we make this whole else-if chain into a switch statement? It also seems rather inefficient to use `getKind()` in every condition.

Reply:
I'll cache getKind().